lib/lightning/work_orders.ex

```diff
@@ -228,15 +228,74 @@ defmodule Lightning.WorkOrders do
         end)
       end)
 
+    do_retry(
+      attempt.work_order,
+      run.input_dataclip,
+      run.job,
+      runs,
+      attrs[:created_by]
+    )
+  end
+
+  def retry(%Attempt{id: attempt_id}, %Run{id: run_id}, opts) do
+    retry(attempt_id, run_id, opts)
+  end
+
+  @spec retry(
+          Attempt.t() | Ecto.UUID.t(),
+          [
+            work_order_option(),
+            ...
+          ]
+          | []
+        ) ::
+          {:ok, Attempt.t()} | {:error, Ecto.Changeset.t(Attempt.t())}
+
```

Comment:
Don't we have the `retry/2` spec further up? Two other things I'm curious about:

1. What happened to the default (`\\ [] `)? 
2. Why the `...`, should any extra options be typed either by union or added to the `work_order_option` type?